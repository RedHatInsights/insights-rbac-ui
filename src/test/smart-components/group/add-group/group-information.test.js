import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import GroupInformation from '../../../../smart-components/group/add-group/group-information';

describe('<GroupInformation />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      formValue: {
        name: 'Foo',
        description: 'This is a test',
      },
      onHandleChange: jest.fn(),
      setIsGroupInfoValid: jest.fn(),
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<GroupInformation {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
