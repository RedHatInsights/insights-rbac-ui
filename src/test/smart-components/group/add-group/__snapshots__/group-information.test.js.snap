// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupInformation /> should render correctly 1`] = `
<GroupInformation
  formValue={
    Object {
      "description": "This is a test",
      "name": "Foo",
    }
  }
  onHandleChange={[MockFunction]}
  setIsGroupInfoValid={[MockFunction]}
>
  <Stack
    hasGutter={true}
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Title
            headingLevel="h4"
            size="xl"
          >
            <h4
              className="pf-c-title pf-m-xl"
            >
              Name and description
            </h4>
          </Title>
        </div>
      </StackItem>
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Form>
            <form
              className="pf-c-form"
              noValidate={true}
            >
              <FormGroup
                fieldId="group-name"
                helperTextInvalid="Required value"
                isRequired={true}
                label="Group name"
                validated="success"
              >
                <div
                  className="pf-c-form__group"
                >
                  <div
                    className="pf-c-form__group-label"
                  >
                    <label
                      className="pf-c-form__label"
                      htmlFor="group-name"
                    >
                      <span
                        className="pf-c-form__label-text"
                      >
                        Group name
                      </span>
                      <span
                        aria-hidden="true"
                        className="pf-c-form__label-required"
                      >
                         
                        *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    className="pf-c-form__group-control"
                  >
                    <ForwardRef
                      aria-describedby="group-name"
                      id="group-name"
                      isRequired={true}
                      name="group-name"
                      onChange={[Function]}
                      type="text"
                      validated="default"
                    >
                      <TextInputBase
                        aria-describedby="group-name"
                        aria-label={null}
                        className=""
                        id="group-name"
                        innerRef={null}
                        isDisabled={false}
                        isLeftTruncated={false}
                        isReadOnly={false}
                        isRequired={true}
                        name="group-name"
                        onChange={[Function]}
                        type="text"
                        validated="default"
                      >
                        <input
                          aria-describedby="group-name"
                          aria-invalid={false}
                          aria-label={null}
                          className="pf-c-form-control"
                          disabled={false}
                          id="group-name"
                          name="group-name"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          readOnly={false}
                          required={true}
                          type="text"
                        />
                      </TextInputBase>
                    </ForwardRef>
                  </div>
                </div>
              </FormGroup>
              <FormGroup
                fieldId="group-description"
                label="Group description"
              >
                <div
                  className="pf-c-form__group"
                >
                  <div
                    className="pf-c-form__group-label"
                  >
                    <label
                      className="pf-c-form__label"
                      htmlFor="group-description"
                    >
                      <span
                        className="pf-c-form__label-text"
                      >
                        Group description
                      </span>
                    </label>
                     
                  </div>
                  <div
                    className="pf-c-form__group-control"
                  >
                    <TextArea
                      aria-label={null}
                      className=""
                      id="group-description"
                      isRequired={false}
                      name="group-description"
                      onChange={[Function]}
                      resizeOrientation="both"
                      type="text"
                      validated="default"
                    >
                      <textarea
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control"
                        id="group-description"
                        name="group-description"
                        onChange={[Function]}
                        required={false}
                        type="text"
                      />
                    </TextArea>
                  </div>
                </div>
              </FormGroup>
            </form>
          </Form>
        </div>
      </StackItem>
    </div>
  </Stack>
</GroupInformation>
`;
